### PR TITLE
chore(NODE-6574): remove `bindings` as a production dependency

### PR DESCRIPTION
**Pull request by baileympearson**
#### What is changing?

This PR removes the dependency on `bindings` and installs the bindings directly from the .node file.

In addition to these tests passing, you can check [mongodb-client-encryption's CI](https://github.com/mongodb-js/mongodb-client-encryption/actions/runs/12035709418?pr=53) for an example of a passing CI run that downloads prebuilt binaries.  (I modified the package.json file to always download prebuilds, you can confirm this by viewing the `npm i` logs).

### Double check the following

- [ ] Ran `npm run format:js && npm run format:rs` script
- [ ] Self-review completed using the [steps outlined here](https://github.com/mongodb/node-mongodb-native/blob/HEAD/CONTRIBUTING.md#reviewer-guidelines)
- [ ] PR title follows the [correct format](https://www.conventionalcommits.org/en/v1.0.0/): `type(NODE-xxxx)[!]: description`
  - Example: `feat(NODE-1234)!: rewriting everything in coffeescript`
- [ ] Changes are covered by tests
- [ ] New TODOs have a related JIRA ticket

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -1,8 +1,10 @@
 'use strict';
-const zstd = require('bindings')('zstd');
+const { resolve } = require('path');
 const { promisify } = require('util');
 const { isUint8Array } = require('util/types');
 
+const zstd = require(resolve(__dirname, '../build/Release/zstd.node'));
```
Comment (nbbeeken) [inline lib/index.js:6]:
```suggestion
const zstd = require('../build/Release/zstd.node');
```

No need to run the same resolution logic that require already has, plus the string literal is desirable to fix issues with bundlers (you can ignore/customize a hard-coded string easily)

Comment (nbbeeken) [inline lib/index.js:6]: What about attempting an import of the Debug version?

**Review by nbbeeken — CHANGES_REQUESTED** [review]

File: lib/index.js (hunk 1 above)
Comment (nbbeeken) [reply lib/index.js]: It's not something to be "set up" you just add a flag:

```sh
npx node-gyp rebuild --debug 
```

I don't think we'd ever ship Debug builds if that's what you mean by "set up". What are you imagining for that?

**Review by nbbeeken — APPROVED** [review]
LGTM

**Review by addaleax — APPROVED** [review]

**Pull request merged**